src/openrct2/actions/MazePlaceTrackAction.hpp

```diff
@@ -71,13 +71,13 @@ DEFINE_GAME_ACTION(MazePlaceTrackAction, GAME_COMMAND_PLACE_MAZE_DESIGN, GameAct
             return res;
         }
 
-        uint8_t baseHeight = _loc.z / 8;
-        uint8_t clearanceHeight = (_loc.z + 32) / 8;
+        auto baseHeight = _loc.z;
+        auto clearanceHeight = _loc.z + (4 * COORDS_Z_STEP);
 
-        int8_t heightDifference = baseHeight - surfaceElement->base_height;
+        auto heightDifference = baseHeight - surfaceElement->GetBaseZ();
         if (heightDifference >= 0 && !gCheatsDisableSupportLimits)
         {
-            heightDifference = heightDifference >> 1;
+            heightDifference = heightDifference / (2 * COORDS_Z_STEP);
```

Comment:
This although correct doesn't show the relationship correctly. The type of `RideData5[RIDE_TYPE_MAZE].max_height` is half small z we should maybe make a explicit constant or preferably refactor this out at some point.

Reply:
I was planning on doing the latter (in another PR).